src/Security/Authentication/Auth0Authenticator.php

```diff
@@ -57,53 +47,28 @@ public static function getSubscribedServices()
 
     public function supports(Request $request): ?bool
     {
-        if ($request->attributes->get('_route') !== $this->loginCheckRoute) {
-            return false;
-        }
-
-        return $request->query->has('code') && $request->query->has('state');
+        return $request->attributes->get('_route') === $this->loginCheckRoute;
     }
 
     public function authenticate(Request $request): PassportInterface
     {
-        if (null === $code = $request->query->get('code')) {
-            throw new AuthenticationException('No oauth code in the request.');
-        }
-
-        if (null === $state = $request->query->get('state')) {
-            throw new AuthenticationException('No state in the request.');
-        }
-
-        if (!$this->get(CsrfTokenManagerInterface::class)->isTokenValid(new CsrfToken('auth0-sso', (string) $state))) {
-            throw new AuthenticationException('Invalid CSRF token');
-        }
-
+        $auth0 = $this->get(Auth0::class);
         try {
             $redirectUri = $this->get(HttpUtils::class)->generateUri($request, $this->loginCheckRoute);
-            $response = $this->get(Authentication::class)->codeExchange((string) $code, $redirectUri);
-            $tokenStruct = \json_decode($response->getBody()->__toString(), true, 512, \JSON_THROW_ON_ERROR);
-        } catch (Auth0Exception $e) {
-            throw new AuthenticationException($e->getMessage(), (int) $e->getCode(), $e);
-        }
 
-        try {
-            // Fetch info from the user
-            $response = $this->get(Authentication::class)->userinfo($tokenStruct['access_token']);
-            /** @var array $userInfo */
-            $userInfo = \json_decode($response->getBody()->__toString(), true, 512, \JSON_THROW_ON_ERROR);
-            $userModel = UserInfo::create($userInfo);
-        } catch (\Exception $e) {
-            throw new AuthenticationException('Could not fetch user info from Auth0', 0, $e);
-        }
+            $auth0->exchange($redirectUri);
+            $userModel = UserInfo::create($auth0->getUser());
+            $userProviderCallback = null;
+            if (null !== $up = $this->get(Auth0UserProviderInterface::class)) {
+                $userProviderCallback = static function () use ($up, $userModel) {
+                    return $up->loadByUserModel($userModel);
+                };
+            }
 
-        $userProviderCallback = null;
-        if (null !== $up = $this->get(Auth0UserProviderInterface::class)) {
-            $userProviderCallback = static function () use ($up, $userModel) {
-                return $up->loadByUserModel($userModel);
-            };
+            return new SelfValidatingPassport(new UserBadge($userModel->getEmail(), $userProviderCallback), [new Auth0Badge($userModel)]);
```

Comment:
We cannot be sure that we have `email`. 

Let's use `user_id` for now and #45 will solve this later.